### PR TITLE
ChangeAndIfToEarlyReturnRector: fix test expectation in if_in_foreach fixture

### PR DESCRIPTION
**Pull request by clxmstaab**
found this error while working on https://github.com/rectorphp/rector/pull/4696

### CODE REVIEW

File: rules/solid/src/Rector/If_/ChangeAndIfToEarlyReturnRector.php [hunk 1]
```diff
@@ -122,9 +122,9 @@ public function refactor(Node $node): ?Node
         $this->addNodesAfterNode($ifs, $node);
         $this->addNodeAfterNode($ifReturn, $node);
 
-        $ifParentReturn = $this->getIfParentReturn($node);
-        if ($ifParentReturn !== null) {
-            $this->removeNode($ifParentReturn);
+        $ifNextReturn = $this->getIfNextReturn($node);
+        if ($ifNextReturn !== null && !$this->isIfInLoop($node)) {
```
Comment (clxmstaab) [inline rules/solid/src/Rector/If_/ChangeAndIfToEarlyReturnRector.php:126]: stripping the `return` is still required for https://github.com/clxmstaab/rector/blob/fix-expect/rules/solid/tests/Rector/If_/ChangeAndIfToEarlyReturnRector/Fixture/only_one_if_before_class_method_return.php.inc

**Comment by clxmstaab** [comment]
should be good to go. besides the actual bugfix I also renamed a method+variables, because they did not reflect the implementation

File: rules/solid/tests/Rector/If_/ChangeAndIfToEarlyReturnRector/Fixture/if_in_foreach.php.inc [hunk 2]
```diff
@@ -42,6 +42,7 @@ class IfInForeachClass
                 return;
             }
             $this->processNameOrIdentifier($node);
+            return;
```
Comment (clxmstaab) [inline rules/solid/tests/Rector/If_/ChangeAndIfToEarlyReturnRector/Fixture/if_in_foreach.php.inc:45]: we are loosing a empty line before the `return;`. not sure this is a problem.

Comment (TomasVotruba) [reply rules/solid/tests/Rector/If_/ChangeAndIfToEarlyReturnRector/Fixture/if_in_foreach.php.inc:45]: Spaces are handled by coding standard tools, not a problem here

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Review by TomasVotruba — APPROVED** [review]

**Pull request merged**